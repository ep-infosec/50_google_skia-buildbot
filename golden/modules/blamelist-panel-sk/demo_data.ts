import { Commit } from '../rpc_types';

export const fakeNow = Date.parse('2020-03-22T00:00:00.000Z');

export const blamelist19: Commit[] = [
  {
    commit_time: 1584835000,
    hash: 'dded3c7506efc5635e60ffb7a908cbe8f1f028f1',
    author: 'Alfa (alfa@example.com)',
    message: 'Update provisioning_profile to unbreak iOS since cert refresh',
    cl_url: '',
    id: '001000001600',
  },
  {
    commit_time: 1584834200,
    hash: '44fc53b7f5d8390d7ae228d5fa8bac0b225ed2b9',
    author: 'Bravo (bravo@example.com)',
    message: 'Add BGR_10A2 support to Ganesh',
    cl_url: '',
    id: '001000001601',
  },
  {
    commit_time: 1584834100,
    hash: 'c50f4e9399cc7ae72b05eb641b50a7cd06d5af82',
    author: 'Charlie (charlie@example.com)',
    message: 'Simplify GrTextBlob::flush',
    cl_url: '',
    id: '001000001602',
  },
  {
    commit_time: 1584834000,
    hash: '9145f784f3261f227846e5b08dc2691a888b113c',
    author: 'Delta (deltadelta@example.com)',
    message: 'Implement D3D copySurface.',
    cl_url: '',
    id: '001000001603',
  },
  {
    commit_time: 1584824000,
    hash: '02b91385f58b84dc14cb6f81478b36d7fcb0d24b',
    author: 'Echo Foxtrot (ef@example.com)',
    message: 'Rename flush -> addOp',
    cl_url: '',
    id: '001000001604',
  },
  {
    commit_time: 1584814000,
    hash: 'a1e90aba441b0e68a07e3ab6660cd673335a76f5',
    author: 'Golf Golf Golf Golf Golf Golf Golf(golf@example.com)',
    message: 'Turn on Vulkan bots for Galaxy S20.',
    cl_url: '',
    id: '001000001605',
  },
  {
    commit_time: 1584804000,
    hash: 'b2b9fbc5675c46cb3af30ccd9b6e20bc57ec04c8',
    author: 'Hotel India (hotel.ind@example.com)',
    message: 'Compare all fields in SkSL::Layout::operator==',
    cl_url: '',
    id: '001000001606',
  },
  {
    commit_time: 1584800000,
    hash: 'f1792cde0b2b7b612cac6aa5b9e7a8ce16b9b2cb',
    author: 'Hotel India (hotel.ind@example.com)',
    message: 'Use constant swizzle syntax in GrDrawVerticesOp',
    cl_url: '',
    id: '001000001607',
  },
  {
    commit_time: 1584780000,
    hash: 'fc75d5c52da617b06c7566c9e389587edde3f284',
    author: 'Juliett Kilo (julia.kilo@example.com)',
    message: '[skottie] Brightness and Contrast effect',
    cl_url: '',
    id: '001000001608',
  },
  {
    commit_time: 1584760000,
    hash: 'e15088801e9722d9a320463554a3a21839213b48',
    author: 'Mike Lima (lima@example.com)',
    message: 'detect failed matrix update in SkDraw::drawAtlas()',
    cl_url: '',
    id: '001000001609',
  },
  {
    commit_time: 1584760000,
    hash: '548de7451e752ce0d6fd842c1bb4f04af4c0afdc',
    author: 'Hotel India (hotel.ind@example.com)',
    message: 'Change Marker IDs to be strings',
    cl_url: '',
    id: '001000001610',
  },
  {
    commit_time: 1584740000,
    hash: '7676f4ecf3ec2f544701465faca1d33af1489822',
    author: 'Zulu (zulu@example.com)',
    message: 'Remove SkFrontBufferedStream',
    cl_url: '',
    id: '001000001611',
  },
  {
    commit_time: 1584720000,
    hash: '4baa7326ccfbfaf66e25d91628378536d4999999',
    author: 'Papa November (pnov@example.com)',
    message: '[infra] Add POC task driver',
    cl_url: '',
    id: '001000001612',
  },
  {
    commit_time: 1584700000,
    hash: 'f5132a05c893a86f8bf26bcf3253985d9973fea2',
    author: 'Quebec (quebec@example.com)',
    message: "Reland \"Optimize GrTessellatePathOp's code to emit inner tri",
    cl_url: '',
    id: '001000001613',
  },
  {
    commit_time: 1584600005,
    hash: '3d599d16ecd8009dfb185522e87c5c1cf8a47057',
    author: 'skia-recreate-skps (skia-recreate-skps@example.com)',
    message: 'Update Go Deps',
    cl_url: '',
    id: '001000001614',
  },
  {
    commit_time: 1584600005,
    hash: '7a0517f8bdc0d168565a2944f0c14db06084384d',
    author: 'skia-autoroll (skia-autoroll@example.com)',
    message: 'Roll third_party/externals/angle2 3cb9c4bee9b3..4395170e6091',
    cl_url: '',
    id: '001000001618',
  },
  {
    commit_time: 1584600003,
    hash: 'd18100afb62401b49e0b3a5c87ed8fa006e8d3a6',
    author: 'skia-autoroll (skia-autoroll@example.com)',
    message: 'Roll ../src 9781ff27c9e9..78824aa9d99f (468 commits)',
    cl_url: '',
    id: '001000001619',
  },
  {
    commit_time: 1584600002,
    hash: '2ac6fa0ae796b7f050bd8e8ad60890427a4ec15e',
    author: 'skia-autoroll (skia-autoroll@example.com)',
    message: 'Roll third_party/externals/swiftshader 60aa34a990fa..2717702',
    cl_url: '',
    id: '001000001620',
  },
  {
    commit_time: 1584600001,
    hash: '667edf14ad72966ec36aa6cd705b98cb7d7eee28',
    author: 'skia-autoroll (skia-autoroll@example.com)',
    message: 'Roll third_party/externals/dawn 00b90ea83262..88f2ec853f80 (',
    cl_url: '',
    id: '001000001621',
  },
];

export const clBlamelist: Commit[] = [
  {
    commit_time: 1584600001,
    hash: '12345',
    author: 'skia-autoroll (skia-autoroll@example.com)',
    message: 'Roll third_party/externals/dawn 00b90ea83262..88f2ec853f80 (',
    cl_url: 'https://skia-review.googlesource.com/12345',
    id: '001000001684',
  },
  {
    commit_time: 1584835000,
    hash: 'dded3c7506efc5635e60ffb7a908cbe8f1f028f1',
    author: 'Alfa (alfa@example.com)',
    message: 'Update provisioning_profile to unbreak iOS since cert refresh',
    cl_url: '',
    id: '',
  },
];

export const nonStandardCommits: Commit[] = [
  {
    id: 'R92-13954.0.0', // These are the ids ChromeOS uses
    hash: '',
    commit_time: 0,
    cl_url: '',
    message: '',
    author: '',
  },
  {
    id: 'R92-13958.0.0',
    hash: '',
    commit_time: 0,
    cl_url: '',
    message: '',
    author: '',
  },
  {
    id: 'R92-13959.0.0',
    hash: '',
    commit_time: 0,
    cl_url: '',
    message: '',
    author: '',
  },
];
