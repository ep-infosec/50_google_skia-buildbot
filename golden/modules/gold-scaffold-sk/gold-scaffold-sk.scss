@import 'node_modules/elements-sk/colors.scss';
@import '../colors';

gold-scaffold-sk {
  font-family: Roboto, Helvetica, Arial, 'Bitstream Vera Sans', sans-serif;

  app-sk {
    --header-background-color: var(--gold);
    --header-font-color: var(--on-primary, var(--white));
    --background: var(--white);

    main {
      // Prevent the app-sk from stretching horizontally indefinitely (for example, dots-sk can
      // become very long). See https://css-tricks.com/preventing-a-grid-blowout/.
      min-width: 0;
    }
  }

  header {
    h1 {
      font-weight: bold;
      font-size: 1.5em;
    }

    .spinner-spacer {
      min-width: 40px;
    }

    spinner-sk {
      width: 24px;
      height: 24px;
      margin: 0;
      border: 8px solid var(--light-orange);
      border-left: 8px solid var(--red);
    }

    last-commit-sk {
      color: var(--white);
      margin-right: 5px;
    }

    login-sk {
      margin-left: auto;
      margin-right: 1em;
      display: block;
      color: var(--white);
    }
  }

  nav {
    display: flex;
    flex-direction: column;
    a {
      padding: 0.5em 0.75em;
      color: var(--black);
      fill: var(--black);
      text-decoration: none;
    }
    span {
      padding: 0 0.5em;
      vertical-align: middle;
    }
  }
}
