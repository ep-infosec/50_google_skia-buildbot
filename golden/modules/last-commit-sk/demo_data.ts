import { StatusResponse } from '../rpc_types';

export const exampleStatusData: StatusResponse = {
  lastCommit: {
    id: 'foxtrota828',
    commit_time: 1598983079,
    hash: 'a8281e31afa9dddfa0764f59128c3a2360c48f49',
    author: 'Foxtrot Delta (foxtrot.delta@example.com)',
    message: 'Mark large_image_changer tests as not flaky (#65033)',
    cl_url: '',
  },
  corpStatus: [{
    name: 'flutter', untriagedCount: 0,
  }],
};
