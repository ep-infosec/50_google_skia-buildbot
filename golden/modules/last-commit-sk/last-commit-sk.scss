@import 'node_modules/elements-sk/colors.scss';

last-commit-sk {
  a {
    color: inherit;
  }
}
