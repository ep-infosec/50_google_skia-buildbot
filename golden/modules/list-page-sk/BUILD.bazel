load("//infra-sk:index.bzl", "karma_test", "sk_demo_page_server", "sk_element", "sk_element_puppeteer_test", "sk_page", "ts_library")

sk_demo_page_server(
    name = "demo_page_server",
    sk_page = ":list-page-sk-demo",
)

sk_element(
    name = "list-page-sk",
    sass_deps = [
        "@npm//:node_modules/elements-sk/themes/color-palette.scss",
    ],
    sass_srcs = ["list-page-sk.scss"],
    sk_element_deps = [
        "//golden/modules/corpus-selector-sk",
        "//golden/modules/query-dialog-sk",
        "//golden/modules/search-controls-sk",
        "//golden/modules/sort-toggle-sk",
    ],
    ts_deps = [
        "//golden/modules:common_ts_lib",
        "//golden/modules:rpc_types_ts_lib",
        "//golden/modules:settings_ts_lib",
        "//infra-sk/modules/ElementSk:index_ts_lib",
        "@npm//common-sk",
        "@npm//elements-sk",
        "@npm//lit-html",
    ],
    ts_srcs = [
        "index.ts",
        "list-page-sk.ts",
    ],
    visibility = ["//visibility:public"],
)

sk_page(
    name = "list-page-sk-demo",
    html_file = "list-page-sk-demo.html",
    sk_element_deps = [
        "//golden/modules/gold-scaffold-sk",
        ":list-page-sk",
    ],
    ts_deps = [
        "//golden/modules/last-commit-sk:demo_data_ts_lib",
        "//golden/modules:demo_util_ts_lib",
        "//golden/modules:settings_ts_lib",
        "//golden/modules:shared_demo_data_ts_lib",
        ":test_data_ts_lib",
        "@npm//fetch-mock",
    ],
    ts_entry_point = "list-page-sk-demo.ts",
)

sk_element_puppeteer_test(
    name = "list-page-sk_puppeteer_test",
    src = "list-page-sk_puppeteer_test.ts",
    sk_demo_page_server = ":demo_page_server",
    deps = [
        "//puppeteer-tests:util_ts_lib",
        "@npm//@types/chai",
        "@npm//chai",
        "@npm//puppeteer",
    ],
)

karma_test(
    name = "list-page-sk_test",
    src = "list-page-sk_test.ts",
    deps = [
        ":list-page-sk",
        ":test_data_ts_lib",
        "//golden/modules:settings_ts_lib",
        "//golden/modules/corpus-selector-sk",
        "//golden/modules/corpus-selector-sk:corpus-selector-sk_po_ts_lib",
        "//golden/modules/query-dialog-sk",
        "//golden/modules/query-dialog-sk:query-dialog-sk_po_ts_lib",
        "//infra-sk/modules:test_util_ts_lib",
        "@npm//@types/chai",
        "@npm//chai",
        "@npm//common-sk",
        "@npm//fetch-mock",
    ],
)

ts_library(
    name = "test_data_ts_lib",
    srcs = ["test_data.ts"],
    visibility = ["//visibility:public"],
    deps = ["//golden/modules:rpc_types_ts_lib"],
)
