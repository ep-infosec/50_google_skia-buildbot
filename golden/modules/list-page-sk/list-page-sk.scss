@import 'node_modules/elements-sk/themes/color-palette.scss';

list-page-sk {
  table {
    width: 95%;
    border-collapse: collapse;

    td {
      border: 1px solid var(--grey-400);
    }

    td.center {
      text-align: center;
    }
  }

  .query_params {
    display: flex;

    > * {
      margin: auto 5px;
    }
  }
}
