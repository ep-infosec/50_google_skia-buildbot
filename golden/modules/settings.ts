/**
 * This file lists helpers for returning global constants in Gold. These should not change once
 * an instance has booted up, so they need not be returned via JSON requests.
 *
 * Settings are expected to be in window.GoldSettings and the functions here are nice helpers
 * for that, so as to demystify "where do these values come from?"
 */

export interface GoldSettings {
  title?: string;
  defaultCorpus?: string;
  baseRepoURL?: string;
}

function getSettings(): GoldSettings | undefined {
  return (window as any).GoldSettings as GoldSettings | undefined;
}

export function title(): string {
  return getSettings()?.title || '';
}

export function defaultCorpus(): string {
  return getSettings()?.defaultCorpus || '';
}

export function baseRepoURL(): string {
  return getSettings()?.baseRepoURL || '';
}

export function testOnlySetSettings(newSettings: GoldSettings) {
  (window as any).GoldSettings = newSettings;
}
