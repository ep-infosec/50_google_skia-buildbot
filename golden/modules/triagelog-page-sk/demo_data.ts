import { TriageLogEntry } from '../rpc_types';

export const triageLogs: TriageLogEntry[] = [
  {
    id: '4275c86b-d64a-ae38-d931-24ea9b94c551',
    name: 'userFour@example.com',
    ts: 1607691600000,
    details: [{
      grouping: { name: 'square', source_type: 'corners' }, digest: 'a09a09a09a09a09a09a09a09a09a09a0', label_before: 'untriaged', label_after: 'negative',
    }],
  },
  {
    id: '734d45d8-555a-aca5-6c55-c45039e43f89',
    name: 'fuzzy',
    ts: 1607685060000,
    details: [{
      grouping: { name: 'square', source_type: 'corners' }, digest: 'a08a08a08a08a08a08a08a08a08a08a0', label_before: 'untriaged', label_after: 'positive',
    }],
  },
  {
    id: 'fe054e2f-822a-7e0c-3dfb-0e9586adffe4',
    name: 'userThree@example.com',
    ts: 1607595010000,
    details: [{
      grouping: { name: 'square', source_type: 'corners' }, digest: 'a07a07a07a07a07a07a07a07a07a07a0', label_before: 'untriaged', label_after: 'positive',
    }],
  },
  {
    id: '65693cef-0220-f0aa-3503-1d5df6548ac9',
    name: 'userThree@example.com',
    ts: 1591877595000,
    details: [{
      grouping: { name: 'circle', source_type: 'round' }, digest: '00000000000000000000000000000000', label_before: 'untriaged', label_after: 'negative',
    }],
  },
  {
    id: 'a23a2b37-344e-83a1-fc71-c72f8071280a',
    name: 'userThree@example.com',
    ts: 1591877594000,
    details: [{
      grouping: { name: 'square', source_type: 'corners' }, digest: 'a03a03a03a03a03a03a03a03a03a03a0', label_before: 'untriaged', label_after: 'positive',
    }],
  },
  {
    id: 'c2b9779e-a0e7-9d48-7c91-0edfa48db809',
    name: 'userOne@example.com',
    ts: 1591518188000,
    details: [{
      grouping: { name: 'square', source_type: 'corners' }, digest: 'a01a01a01a01a01a01a01a01a01a01a0', label_before: 'untriaged', label_after: 'positive',
    }, {
      grouping: { name: 'square', source_type: 'corners' }, digest: 'a02a02a02a02a02a02a02a02a02a02a0', label_before: 'untriaged', label_after: 'positive',
    }],
  },
  {
    id: 'f9adaa96-df23-2128-2120-53ea2d57536b',
    name: 'userTwo@example.com',
    ts: 1591517708000,
    details: [{
      grouping: { name: 'triangle', source_type: 'corners' }, digest: 'b04b04b04b04b04b04b04b04b04b04b0', label_before: 'positive', label_after: 'negative',
    }],
  },
  {
    id: '931323d9-926d-3a24-0350-6440a54d52cc',
    name: 'userTwo@example.com',
    ts: 1591517707000,
    details: [{
      grouping: { name: 'triangle', source_type: 'corners' }, digest: 'b04b04b04b04b04b04b04b04b04b04b0', label_before: 'untriaged', label_after: 'positive',
    }],
  },
  {
    id: '1d35d070-9ec6-1d0a-e7bd-1184870323b3',
    name: 'userTwo@example.com',
    ts: 1591517704000,
    details: [{
      grouping: { name: 'triangle', source_type: 'corners' }, digest: 'b03b03b03b03b03b03b03b03b03b03b0', label_before: 'untriaged', label_after: 'negative',
    }],
  },
  {
    id: 'fbbe2efb-5fc0-bd3c-76fa-b52714bad960',
    name: 'userOne@example.com',
    ts: 1591517383000,
    details: [{
      grouping: { name: 'triangle', source_type: 'corners' }, digest: 'b01b01b01b01b01b01b01b01b01b01b0', label_before: 'untriaged', label_after: 'positive',
    }, {
      grouping: { name: 'triangle', source_type: 'corners' }, digest: 'b02b02b02b02b02b02b02b02b02b02b0', label_before: 'untriaged', label_after: 'positive',
    }],
  },
  {
    id: '94a63df2-33d3-97ad-f4d7-341f76ff8cb6',
    name: 'userOne@example.com',
    ts: 1591517350000,
    details: [{
      grouping: { name: 'circle', source_type: 'round' }, digest: 'c01c01c01c01c01c01c01c01c01c01c0', label_before: 'untriaged', label_after: 'positive',
    }, {
      grouping: { name: 'circle', source_type: 'round' }, digest: 'c02c02c02c02c02c02c02c02c02c02c0', label_before: 'untriaged', label_after: 'positive',
    }],
  },

];
