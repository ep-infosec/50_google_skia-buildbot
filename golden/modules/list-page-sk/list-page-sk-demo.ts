import './index';
import '../gold-scaffold-sk';
import fetchMock from 'fetch-mock';
import { delay } from '../demo_util';
import { manyParams } from '../shared_demo_data';
import { testOnlySetSettings } from '../settings';
import { sampleByTestList } from './test_data';
import { exampleStatusData } from '../last-commit-sk/demo_data';
import { ListPageSk } from './list-page-sk';
import { GoldScaffoldSk } from '../gold-scaffold-sk/gold-scaffold-sk';

testOnlySetSettings({
  title: 'Testing Gold',
  defaultCorpus: 'gm',
  baseRepoURL: 'https://github.com/flutter/flutter',
});

fetchMock.get('/json/v2/paramset', delay(manyParams, 100));
fetchMock.get('glob:/json/v2/list*', delay(sampleByTestList, 100));
fetchMock.get('/json/v2/trstatus', JSON.stringify(exampleStatusData));

// By adding these elements after all the fetches are mocked out, they should load ok.
const newScaf = new GoldScaffoldSk();
newScaf.testingOffline = true;
// Make it the first element in body.
document.body.insertBefore(newScaf, document.body.childNodes[0]);
newScaf.appendChild(new ListPageSk());
