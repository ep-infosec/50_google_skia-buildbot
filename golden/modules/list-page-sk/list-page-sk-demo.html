<!DOCTYPE html>
<html>
  <head>
    <title>list-page-sk</title>
    <meta charset="utf-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  </head>
  <body class="body-sk">
    <!-- demo page code will insert
<gold-scaffold-sk testing_offline>
  <list-page-sk></list-page-sk>
</gold-scaffold-sk>
here -->
  </body>
</html>
